test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -1444,6 +1444,35 @@ Describe "Type inference Tests" -tags "CI" {
         )
         $null = [AstTypeInference]::InferTypeOf($FoundAst)
     }
+
+    It 'Infers type of command with all streams redirected to Success stream' {
+        $res = [AstTypeInference]::InferTypeOf( { Get-PSDrive *>&1 }.Ast)
+        $ExpectedTypeNames = @(
+            [ErrorRecord].FullName
+            [WarningRecord].FullName
+            [VerboseRecord].FullName
+            [DebugRecord].FullName
+            [InformationRecord].FullName
+            [PSDriveInfo].FullName
+        ) -join ';'
+        $res.Name -join ';' | Should -Be $ExpectedTypeNames
+    }
+
+    It 'Infers type of command with success stream redirected' {
+        $res = [AstTypeInference]::InferTypeOf( { Get-PSDrive *>&1 1>$null }.Ast)
```

Comment:
How does this works with #20381?

Reply:
Great question! The type inference code doesn't currently handle redirected variables so my changes in this PR don't have any impact on that. As you know, I have a pending PR for improving the variable type inference so I will update the new visitor to also take variable redirections into account. I don't need to make any changes in this PR to accommodate that.